File: comparisons/mapreduce/mapreduce.rb

somaDosParesAoQuadrado = 
    (1..100)
    .find_all {|x|      x % 2 == 0}
    .map      {|x|      x ** 2}
    .inject   {|acc, x| acc + x}

puts somaDosParesAoQuadrado
 

File: comparisons/mapreduce/mapreduce.hs

somaDosParesAoQuadrado = 
    foldr (\acc x -> acc + x) 0
    $ map (\x -> x ^ 2)
    $ filter (\x -> x `mod` 2 == 0)
    $ [1..100] 

main = print somaDosParesAoQuadrado
